src/utils.rs

```diff
@@ -152,6 +152,66 @@ pub fn ensure_procfs(path: &Path) -> Result<()> {
     Ok(())
 }
 
+pub fn secure_join(rootfs: &Path, unsafe_path: &Path) -> Result<PathBuf> {
+    
+    let mut clean_path = PathBuf::from(rootfs);
+    let mut path = PathBuf::from(unsafe_path);
+    let mut p = PathBuf::new();
+
+    let mut part = path.iter();
+    let mut i = 0;
+
+    loop {
+        if i > 255 {
+            // Dereference too many symlinks, may be infinite loop
+            break;
+        }
+
+        let part_path;
+        match part.next() {
+            Some(part) => {
+                part_path = PathBuf::from(part);
+            },
+            None => {
+                break;
+            }
+        }
+
+        if !part_path.is_absolute() {
+            if part_path.starts_with("..") {
+                p.pop();
+            } else {
+                // check if symlink then dereference
+                let metadata = match PathBuf::from(&clean_path).join(&p).join(&part_path).symlink_metadata() {
```

Comment:
We can error out here as well?

Reply:
Do you mean error out if the metadata is not obtained? The ```symlink_metadata``` will throw error if the file does not exists and in my ```test_secure_join```, the paths I'm using don't exist. So, those tests will fail if I error out here. One way is to create those paths like in ```test_secure_join_symlink```. What do you think?

Reply:
I see. Then let's add a comment here to explain. Also can be just ignore the specific error and error out when it's the other error? I assume we only want to return `None` for the case symblink doesn't exist.